bloom-housing/bloom
# chore: add max lines linting rules and refactor

## Issue Overview

This PR addresses #2337 


## Description

Configures eslint with two new rules: max-lines-per-function and max-lines (per file). The intention is to encourage breaking up large single functions (sometimes with nested functions) into smaller discrete functions and perhaps additional files, so code blocks are simpler to maintain and easier to unit test—and possibly improve performance or memory usage as well.

FYI, the numbers I chose for the rules was after tweaking the values a few different times to find a sweet spot between not enough stuff/too much stuff getting caught in our existing codebase. Also the way React Hook Form-related components get written ends up with really long functions much of the time, so I'm not sure how much we can mitigate against that, but it does seems like there's some room for improvement in a few places.

## How Can This Be Tested/Reviewed?

Obviously we can tell if the linter & tests all run as expected, but in the case of refactored files we should try to QA the pages or API endpoints which were modified.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
